### PR TITLE
[action] add AccessList input to NewExecution()

### PR DESCRIPTION
**Pull request by dustinxie**
# Description
1. commit https://github.com/iotexproject/iotex-core/commit/a4e5e6a3a36080a96255afca1e3e9168393e9b20 adds access list, which is feature to be enabled in 190 (London EVM)
2. need to add AccessList as input to NewExecution()

## Type of change
Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
- [x] make test

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by CoderZhi — COMMENTED** [review]
Is `accesslist` only valid for `contract creation` case?

**Comment by dustinxie** [comment]
> Is `accesslist` only valid for `contract creation` case?

it is for normal execution, purpose is to charge less gas, if you know ahead the value of keys (like storage keys of account balance) the contract will access (and pass those with input)

my understanding, it should be extreme rare that `contract creation` will use this feature (since the contract is not created yet, should be no `GetState()` calls)

File: action/execution.go [hunk 1]
```diff
@@ -50,6 +50,7 @@ func NewExecution(
 	gasLimit uint64,
 	gasPrice *big.Int,
 	data []byte,
+	list types.AccessList,
```
[the remark this answers is not in the document]
Comment (dustinxie) [reply action/execution.go]: updated

File: api/grpcserver.go [hunk 2]
```diff
@@ -576,37 +575,22 @@ func (svr *GRPCServer) TraceTransactionStructLogs(ctx context.Context, in *iotex
 	if err != nil {
 		return nil, err
 	}
-	exec, ok := actInfo.Action.Core.Action.(*iotextypes.ActionCore_Execution)
-	if !ok {
-		return nil, status.Error(codes.InvalidArgument, "the type of action is not supported")
-	}
-	callerAddr, err := address.FromString(actInfo.Sender)
+	act, err := (&action.Deserializer{}).ActionToSealedEnvelope(actInfo.Action)
```
Comment (dustinxie) [inline api/grpcserver.go:578]: this is a more organic way of converting protobuf action

File: action/execution.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Liuhaai) [reply action/execution.go:75]: > updated

NewAccessListExecution would be better. Newexec seems like a default tx. However, legacy exec is used in most of tome

**Review by Liuhaai — APPROVED** [review]

File: action/execution.go [hunk 3]
```diff
@@ -64,6 +64,30 @@ func NewExecution(
...
+func NewExecution(
+	contractAddress string,
+	nonce uint64,
+	amount *big.Int,
+	gasLimit uint64,
+	gasPrice *big.Int,
+	data []byte,
+	list types.AccessList,
+) (*Execution, error) {
+	return &Execution{
```
Comment (CoderZhi) [inline action/execution.go:77]: https://github.com/iotexproject/iotex-core/pull/3405/files/52b5039dc0fad1e75aad6a6801ed5cfa95d7a05d..5d1ec60d88e831002efcb4a761416e327b4e87ea#r887155645

Of cause, I prefer your current implementation.

File: action/execution.go [hunk 4]
```diff
@@ -42,8 +42,8 @@ type Execution struct {
 	accessList types.AccessList
 }
 
-// NewExecution returns a Execution instance
-func NewExecution(
+// NewLegacyExecution returns a legacy Execution instance (w/o access list)
+func NewLegacyExecution(
```
Comment (CoderZhi) [inline action/execution.go:46]: it is not a legacy execution, but an execution with `accesslist == nil`. I may call it `NewExecution`, and call the new one `NewExecutionWithAccessList`

**Review by CoderZhi — APPROVED** [review]

File: action/execution.go (hunk 4 above)
Comment (dustinxie) [reply action/execution.go:46]: ok, this is also haixiang's comment

**Pull request merged**